### ISSUE TITLE
bug: annotate_timestamp method for ds using "code" unit_system

### ISSUE THREAD

**Issue by neutrinoceros** [yt-project/yt#2432]
## Bug report

**Bug summary**
the "code" unit system is correctly being used for x, y, z labels (since #2354) but not for timestamps.

**Code for reproduction**
```python
import yt
ds = yt.load("amrvac/bw_2d0000.dat", unit_system="code")
p = yt.plot_2d(ds, "density")
p.annotate_timestamp()
```

**Actual outcome**
![code_timestamp_Slice_phi_density](https://user-images.githubusercontent.com/14075922/73372404-6fc74480-42b7-11ea-8076-3ac30645a388.png)

**Expected outcome**

the time stamp should use "code time" as a unit.

**Comment by munkm** [comment]
I'm working on a fix for this. Should be ready soon.

**Comment by munkm** [comment]
Resolved via yt-project/yt#2435
